crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -204,13 +214,13 @@ impl CheckpointHandler {
             last_epoch: Some(IndexedEpochInfo::from_end_of_epoch_data(
                 system_state_summary.clone(),
                 checkpoint_summary,
-                &event,
+                epoch_event_opt.as_ref(),
                 network_tx_count_prev_epoch,
             )),
             new_epoch: IndexedEpochInfo::from_new_system_state_summary(
                 system_state_summary,
                 checkpoint_summary.sequence_number + 1, // first_checkpoint_id
-                Some(&event),
+                epoch_event_opt.as_ref(),
```

Comment:
rather than make this param optional, what are your thoughts on assigning a default to epoch_event_opt and passing that to IndexedEpochInfo::from_end_of_epoch_data. With the check and assert, we can only have an Option iff we are in safe mode, because the only other time this event is None is at genesis, which is handled separately

Reply:
I started it that way actually https://github.com/MystenLabs/sui/pull/20015#discussion_r1815590314
this is a patch for 1-25 only, and the one merged to main is here https://github.com/MystenLabs/sui/pull/20015